stdlib/public/core/CMakeLists.txt

```diff
@@ -434,6 +434,15 @@ if(SWIFT_SHOULD_BUILD_EMBEDDED_STDLIB)
   set(SWIFT_STDLIB_ENABLE_OBJC_INTEROP OFF)
   set(SWIFT_STDLIB_ENABLE_VECTOR_TYPES OFF)
 
+  # Under Embedded Swift, all stdlib APIs should be available always. Replace
+  # all availability macros with 10.13.
+  set(SWIFT_STDLIB_AVAILABILITY_DEFINITIONS_EMBEDDED)
+  foreach(def ${SWIFT_STDLIB_AVAILABILITY_DEFINITIONS})
+    string(REGEX REPLACE ":.*" ":macOS 10.13" replaced "${def}")
```

Comment:
It feels a bit weird to me to only specify `macOS` availability in the replacement. Either there should be no platforms with explicit availability (this may not be supported by the existing availability macros implementation), or each of the platforms listed in the existing macro should be lowered to their minimum. Embedded Swift should be able to interoperate with each of the platform equivalently, right?

Reply:
I think I'll need some educating here :) What does `@available(macOS 10.13, *)` mean around availability on iOS and other OS's?

Reply:
The `*` syntax is a reminder that you're not placing any constraints on any other platforms with that attribute, it just applies to `macOS`. When compiling for iOS, for example, `@available(macOS 10.13, *)` is ignored entirely. It's the same thing as if you wrote no `@available` attributes at all.

Reply:
Ok, makes sense!

Reply:
(hopefullly) fixed now

Reply:
LGTM